packages/@aws-cdk/aws-codestar/test/integ.github.ts

```diff
@@ -0,0 +1,17 @@
+import { Bucket } from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import { GitHubRepository } from '../lib';
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'GitHubDemo');
+
+new GitHubRepository(stack, 'GitHubRepo', {
+  owner: 'foo',
+  gitHubRepositoryName: 'bar',
+  accessToken: cdk.SecretValue.secretsManager('my-github-token', {
+    jsonField: 'token',
+  }),
+  bucket: Bucket.fromBucketName(stack, 'Bucket', 'bucket-name'),
```

Comment:
> This should be a new S3 Bucket (otherwise, this test will fail to deploy).

I feel this integ test is very challenging as I need to pass in real GitHub token in my Secrets Manager as a real Secret, creating a real S3 bucket and upload a real .zip into that bucket so it can deploy in my development environment.

However, I can't just submit this test like that and I have to make it agnostic, which means I probably have to create a new bucket and secret and upload a zip into that bucket and eventually this deployment test would still fail because the github token and account are not real.

What would you suggest in this scenario?

Reply:
Skip the integ test :)